File: solutions/javascript/resistor-color/1/resistor-color.js
// @ts-check

// Define the COLORS array first, as it's a constant and will be used by colorCode.
// The order of colors in this array corresponds to their numerical value (index).
export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];

/**
 * Returns the numerical value associated with a given resistor color band.
 *
 * @param {string} color The color name (lowercase).
 * @returns {number} The numerical value for the color.
 * @throws {Error} If the color is not found in the COLORS list.
 */
export const colorCode = (color) => {
  const index = COLORS.indexOf(color);

  if (index === -1) {
    // If the color is not found in the array, it's an invalid input.
    // Throw an error, although the exercise description doesn't explicitly require it,
    // it's good practice for invalid inputs.
    throw new Error(`Invalid color: ${color}. Color not found in the resistor color code.`);
  }

  return index;
};

File: solutions/java/resistor-color/1/src/main/java/ResistorColor.java
class ResistorColor {
    
    // Define the colors in order of their numeric value (0-9)
    private static final String[] COLORS = {
        "black", "brown", "red", "orange", "yellow", 
        "green", "blue", "violet", "grey", "white"
    };

    int colorCode(String color) {
        // Find the index of the color in the array, which corresponds to its value
        for (int i = 0; i < COLORS.length; i++) {
            if (COLORS[i].equals(color)) {
                return i;
            }
        }
        // Should not happen with valid input, but good practice to handle it
        throw new IllegalArgumentException("Invalid color: " + color);
    }

    String[] colors() {
        // Return the full list of colors
        return COLORS.clone(); // Return a copy to prevent external modification
    }
}